### ISSUE TITLE
rename ChartController

### ISSUE THREAD

**Issue by ldalegandar** [jettro/c3-angular-directive#122]
I suggest renaming the internal controller to something more specific - e.g c3ChartController.

I was testing c3-angular for the first time and so I used the generic ChartController name without realising it was already being used by c3-angular. This caused a naming conflict and the whole app fell over with very little helpful errors (i.e. angular didn't say there was any namespace conflict, just that a bunch of functions weren't found). It wasn't until I started digging in the source file and API that I realised the name conflict was to blame.

Given that people are likely to use generic names like this in their apps, I would strongly suggest you use a much more explicit name for the internal controller - or make it much clearer in the documentation.

**Comment by jettro** [comment]
Good point, will do some testing to see if this breaks something.

**Comment by ldalegandar** [comment]
I can confirm that doing a replace all in c3-angular.js to replace 'ChartController' with 'c3ChartController' didn't break anything for me (it was the first thing I did when I found out). But it would probably need more testing.

You'll obviously need to update the API to reflect the change and if anyone is using the ChartController in their app then changing it could break functionality? Personally I think it's a good change going forward, but completely up to you.

**Comment by MDBox** [comment]
I agree, it would be nice if the controller was renamed or a clear warning not to use 'ChartController' in your app.  I just did the same thing and couldn't figure out why this worked in one controller but not that other.
